Frameworks/Foundation/NSOperationQueue.mm

```diff
@@ -196,6 +196,12 @@ - (NSOperationQueue*)_doMainWork {
         didWork = FALSE;
 
         for (int i = 0; i < NSOperationQueuePriority_Count; i++) {
+            [priv->suspendedCondition lock];
+            while (priv->isSuspended) {
+                [priv->suspendedCondition wait];
+            }
```

Comment:
Cool, makes sense. I remember a similar situation some time ago where I was told not to use a while loop on a condition. I was just curious as to the reasoning. Thanks. :)

---
In reply to: [69806286](https://github.com/Microsoft/WinObjC/pull/613#discussion_r69806286) [](ancestors = 69806286,69805759)